File: Desafio 3/desafio3.py
class Heroi:
    def __init__(self, nome, idade, tipo):
        self.nome = nome
        self.idade = idade
        self.tipo = tipo

    def atacar(self):
        if self.tipo == "mago":
            ataque = "magia"
        elif self.tipo == "guerreiro":
            ataque = "espada"
        elif self.tipo == "monge":
            ataque = "artes marciais"
        elif self.tipo == "ninja":
            ataque = "shuriken"
        else:
            ataque = "ataque indefinido"

        print(f"O {self.tipo} atacou usando {ataque}.")


heroi1 = Heroi("Anagorn", 30, "guerreiro")
heroi2 = Heroi("Gandalf", 150, "mago")

heroi1.atacar()
heroi2.atacar()

File: Desafio 3/desafio3.js
class Heroi {
    constructor(nome, idade, tipo) {
        this.nome = nome;
        this.idade = idade;
        this.tipo = tipo;
    }

    atacar() {
        let ataque;

        switch (this.tipo) {
            case "mago":
                ataque = "magia";
                break;
            case "guerreiro":
                ataque = "espada";
                break;
            case "monge":
                ataque = "artes marciais";
                break;
            case "ninja":
                ataque = "shuriken";
                break;
            default:
                ataque = "ataque indefinido";
        }

        console.log(`O ${this.tipo} atacou usando ${ataque}.`)
    }
}

let heroi1 = new Heroi("Aragorn", 30, "guerreiro");
let heroi2 = new Heroi("Gandalf", 150, "mago");

heroi1.atacar();
heroi2.atacar();